src/app/subnets/components/ReservedRanges/useReservedRangesColumns/useReservedRangesColumns.tsx

```diff
@@ -0,0 +1,108 @@
+import { useMemo } from "react";
+
+import type { ColumnDef } from "@tanstack/react-table";
+
+import { Labels } from "../ReservedRanges";
+
+import SubnetLink from "@/app/base/components/SubnetLink";
+import TableActions from "@/app/base/components/TableActions";
+import type { SetSidePanelContent } from "@/app/base/side-panel-context";
+import {
+  SubnetActionTypes,
+  SubnetDetailsSidePanelViews,
+} from "@/app/subnets/views/SubnetDetails/constants";
+
+export type ReservedRangesTableData = {
+  id: number | string;
+  ipRangeId?: number;
+  subnet: number | null;
+  startIp: string;
+  endIp: string;
+  owner: string;
+  type: string;
+  comment: string;
+};
+
+export type ReservedRangesColumnsDef = ColumnDef<
+  ReservedRangesTableData,
+  Partial<ReservedRangesTableData>
+>;
+
+const useReservedRangesColumns = (
+  showSubnetColumn: boolean,
+  setSidePanelContent: SetSidePanelContent
+): ReservedRangesColumnsDef[] => {
+  return useMemo((): ReservedRangesColumnsDef[] => {
```

Comment:
Move the context call in here.
```suggestion
const useReservedRangesColumns = (
  showSubnetColumn: boolean
): ReservedRangesColumnsDef[] => {
  const { setSidePanelContent } = useSidePanel();
  return useMemo((): ReservedRangesColumnsDef[] => {
```